rpc/core/blocks.go

```diff
@@ -168,6 +195,36 @@ func BlockSearch(
 	orderBy string,
 ) (*ctypes.ResultBlockSearch, error) {
 
+	results, err := blocksByQuery(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+
+	// sort results (must be done before pagination)
+	err = sortBlocks(results, orderBy)
+	if err != nil {
+		return nil, err
+	}
+
+	// paginate results
+	totalCount := len(results)
+	perPage := validatePerPage(perPagePtr)
+
+	page, err := validatePage(pagePtr, perPage, totalCount)
+	if err != nil {
+		return nil, err
+	}
+
+	skipCount := validateSkipCount(page, perPage)
+	pageSize := tmmath.MinInt(perPage, totalCount-skipCount)
+
+	apiResults := fetchBlocks(results, pageSize, skipCount)
+
+	return &ctypes.ResultBlockSearch{Blocks: apiResults, TotalCount: totalCount}, nil
+}
+
+// blocksByQuery queries for blocks and returns the whole result list.
+func blocksByQuery(ctx *rpctypes.Context, query string) ([]int64, error) {
```

Comment:
Nice catch :+1: The thing is, `blockByQuery` is used in two ways:
- One that uses pagination and stuff. Then it is guaranteed to only return a defined number of results.
- Data commitment, and it doesn't take any pagination.

However, we can add a pre-condition, in the data commitment function, that checks over the number of blocks needed, `1000` for example, and throws in case the query exceeds it.
I think, yes, it makes sense to implement it. Except that for the limit, we should maybe ask @adlerjohn as he might have more visibility on the batches size etc. What do you think @evan-forbes ?

Reply:
It's definitely not a necessary feature, so we can merge without it. I would only add it if its straight forward, like if we could add a line to the client code to quickly check the query. If not, then I think we should keep moving and fix it if it actually becomes a problem